### ISSUE TITLE
Exception after upgrading to latest aws-xray-sdk-core (2.3.6)

### ISSUE THREAD

**Issue by onionhammer** [aws/aws-xray-sdk-node#183]
"aws-xray-sdk-core": "^2.3.6",

Looks like a null reference exception deep in the stack

![image](https://user-images.githubusercontent.com/969938/64806905-c0810f80-d559-11e9-9381-550cb67ec7b3.png)

**Comment by willarmiros** [comment]
Hmm, the only change that was made in the 2.3.6 patch was this [bug fix](https://github.com/aws/aws-xray-sdk-node/commit/42e4aa8ce89418e7dff5beef323c900760a103df) related to logging in Lambda environments. 

I'm still happy to take a look at a code snippet if you're still having issues.

**Comment by onionhammer** [comment]
@willarmiros Or we could just fix the null reference exception and throw a more helpful error

https://github.com/aws/aws-xray-sdk-node/blob/master/packages/core/lib/context_utils.js#L89

```js
      var ns = cls.getNamespace(NAMESPACE);
      if (!ns) throw new Error("Namespace must be configured in cls mode");
      var segment = ns.get(SEGMENT);


```

**Comment by willarmiros** [comment]
We are always open to PRs to better inform our customers of the errors they are experiencing. If you make one, please make it against the `2.x` [branch](https://github.com/aws/aws-xray-sdk-node/tree/2.x), as that is where we are making releases until the next major version is released.

**Comment by willarmiros** [comment]
Closing since it was fixed in 2.4.0.
